### PR TITLE
feat(E621): add privacy mode

### LINKED ISSUE #5328: [Presence | Feature Request] e621.net
### Discussed in https://github.com/PreMiD/Presences/discussions/4649

<div type='discussions-op-text'>

<sup>Originally posted by **Mr-Foxtrot** November 29, 2020</sup>
**What's the name of the service?**
e621.net

**Is your feature request related to a problem? Please describe.**
Well, I am aware the presence already exists. What I came here to request however was a settings page for it, as unlike other presences which lets you hide your search results or what page you are specifically viewing (an example is the Furaffinity rich presence) the e621 rich presence doesn't have any settings. So anyone who views your discord profile knows exactly what you are searching on the site, what post you are on, etc. A option to hide those details would be great and having it just show "Browsing e621" would be lovely. 

I don't believe a lot of people want to know exactly what you are searching up on e6 =/</div>

### PR DESCRIPTION
**Pull request by DarkVillager**
Resolves #5328
**Describe the changes in this pull request:**
- Added Privacy Mode
- Changed Buttons not to show when Privacy mode is enabled
- Added Check to use the right button text
- Changed buttons text

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

</details>

### CODE REVIEW

File: websites/E/e621/presence.ts [hunk 1]
```diff
@@ -8,194 +8,222 @@ presence.on("UpdateData", async () => {
 			largeImageKey: "logo"
 		},
 		path = document.location.pathname,
+		privacy = await presence.getSetting<boolean>("privacy"),
```
Comment (Bas950) [inline websites/E/e621/presence.ts:11]: promise.all

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/E/e621/presence.ts [hunk 2]
```diff
@@ -8,194 +8,224 @@ presence.on("UpdateData", async () => {
...
-					label: "View Post",
-					url: document.location.href
+		[privacy, buttons] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("buttons")
+		]);
+	if (!privacy) {
+		if (path === "/") {
+			search = document.querySelector("#tags");
+			if (search.value !== "") {
```
Comment (Bas950) [inline websites/E/e621/presence.ts:18]:
```suggestion
			if (search.value) {
```

File: websites/E/e621/presence.ts [hunk 3]
```diff
@@ -8,194 +8,224 @@ presence.on("UpdateData", async () => {
...
+				presenceData.details = "Searching For:";
+				presenceData.state = search.value;
+			} else presenceData.details = "Viewing the homepage";
+		} else if (path.includes("posts")) {
+			search = document.querySelector("#tags");
+			if (
+				search.value !== "" &&
```
Comment (Bas950) [inline websites/E/e621/presence.ts:25]:
```suggestion
				search.value &&
```

File: websites/E/e621/presence.ts [hunk 4]
```diff
@@ -8,194 +8,224 @@ presence.on("UpdateData", async () => {
...
+					.split("created by");
+				presenceData.details = split[0];
+				presenceData.state = `Created By: ${split[1]}`;
+			} else {
+				presenceData.details = "All Posts";
+				delete presenceData.state;
+			}
+			if (buttons) {
+				if (
+					(search.value !== "" &&
```
Comment (Bas950) [inline websites/E/e621/presence.ts:48]:
```suggestion
					(search.value &&
```

File: websites/E/e621/presence.ts [hunk 5]
```diff
@@ -8,194 +8,224 @@ presence.on("UpdateData", async () => {
...
+					{
+						label: "View Profile",
+						url: document.location.href
+					}
+				];
+			}
+		} else if (path.includes("artists")) {
+			search = document.querySelector("#search_any_name_matches");
+			if (!search) search = document.querySelector("#quick_search_name");
+			if (search.value !== "") {
```
Comment (Bas950) [inline websites/E/e621/presence.ts:106]:
```suggestion
			if (search.value) {
```

File: websites/E/e621/presence.ts [hunk 6]
```diff
@@ -8,194 +8,224 @@ presence.on("UpdateData", async () => {
...
+			if (buttons) {
+				presenceData.buttons = [
+					{
+						label: "View Tags",
+						url: document.location.href
+					}
+				];
+			}
+			search = document.querySelector("#search_name_matches");
+			if (search.value !== "") {
```
Comment (Bas950) [inline websites/E/e621/presence.ts:134]:
```suggestion
			if (search.value) {
```

File: websites/E/e621/presence.ts [hunk 7]
```diff
@@ -8,194 +8,224 @@ presence.on("UpdateData", async () => {
...
+			if (buttons) {
+				presenceData.buttons = [
+					{
+						label: "View Pools",
+						url: document.location.href
+					}
+				];
+			}
+			search = document.querySelector("#search_name_matches");
+			if (search.value !== "") {
```
Comment (Bas950) [inline websites/E/e621/presence.ts:174]:
```suggestion
			if (search.value) {
```

File: websites/E/e621/presence.ts [hunk 8]
```diff
@@ -8,194 +8,224 @@ presence.on("UpdateData", async () => {
...
+			if (buttons) {
+				presenceData.buttons = [
+					{
+						label: "View Forum Post",
+						url: document.location.href
+					}
+				];
+			}
+			search = document.querySelector("#quick_search_body_matches");
+			if (search.value !== "") {
```
Comment (Bas950) [inline websites/E/e621/presence.ts:206]:
```suggestion
			if (search.value) {
```

**Review by Bas950 — DISMISSED** [review]
empty strings are falsy on their own

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**